### DISCUSSION TITLE
I can add our board ESP8266 based into https://arduino.esp8266.com/stable/package_esp8266com_index.json

### DISCUSSION THREAD

**Discussion by Bighoneypot** [esp8266/Arduino#7882] [Q&A]
Hello everyone, I would like to know how I can add my board "Agrumino Lemon" esp8266 to install it on Arduino IDE with json https://arduino.esp8266.com/stable/package_esp8266com_index.json

I want our board to be inserted together with the other esp8266 boards

Thanks, I hope to receive your support.

[Board.txt](https://github.com/esp8266/Arduino/files/5988231/Board.txt)

**Comment by d-a-v** [comment]
Instructions are written in the beginning of the file that you have posted.
Did you try that ?
Is there a website about your board ?

**Reply by Bighoneypot** [reply]
> Instructions are written in the beginning of the file that you have posted.
> Did you try that ?
> Is there a website about your board ?

@d-a-v Hi, thank tou for reply.
Yes, our website is www.lifely.cc and now is possible to buy on Amazon and directly  from website.

**Comment by d-a-v** [comment]
> Based on a ESP8266 WiFi Chip. The official programming environment is Arduino, thanks to the open-source project ESP8266.

esp8266/Arduino is open source.
But we'd surely love that esp8266 were open-source :)

**Reply by Bighoneypot** [reply]
Our device is open source to programming but for the moment we don't release schematics.

**Reply by d-a-v** [reply]
I was not refering to your schematics but on the false statement on your web page under the `Key Features of Agrumino`, section `WiFi`:
> Based on a ESP8266 WiFi Chip. The official programming environment is Arduino, thanks to the open-source project ESP8266.

To our dismay, "project ESP8266" is not open source.
But esp8266/Arduino surely is.

**Reply by Bighoneypot** [reply]
@d-a-v this is a automatic translation error from italian language. Now I fix this. Sorry for this mistake. To avoid mistakes, I have deleted it.

Is possible to add my board on your json https://arduino.esp8266.com/stable/package_esp8266com_index.json ?? Thank you

**Reply by d-a-v** [reply]
> Is possible to add my board on your json https://arduino.esp8266.com/stable/package_esp8266com_index.json ?? Thank you

As stated in the file you posted in OP:

> *Do not create pull-requests for this file only, CI will not accept them.
You *must* edit/modify/run boards.txt.py to regenerate boards.txt.
All modified files after running with option "--allgen" must be included in the pull-request.*

You must edit the file `tools/boards.txt.py` and add your board in there. The description will go to this file too. Everything will be updated by running it with the option `--allgen`, including `boards.txt`.. Then you'll have to make a pull-request with all the modified files.

**Comment by Bighoneypot** [comment]
I need to clone https://github.com/esp8266/Arduino ??? Sorry, but for me it
is a new procedure

Il giorno mar 16 feb 2021 alle ore 14:41 david gauchard <
notifications@github.com> ha scritto:

> Is possible to add my board on your json
> https://arduino.esp8266.com/stable/package_esp8266com_index.json ?? Thank
> you
>
> As stated in the file you posted in OP:
>
>
>
> *Do not create pull-requests for this file only, CI will not accept them.
> You must edit/modify/run boards.txt.py <http://boards.txt.py> to regenerate
> boards.txt. All modified files after running with option "--allgen" must be
> included in the pull-request.*
>
> You must edit the file tools/boards.txt.py and add your board in there.
> The description will go to this file too. Everything will be updated by
> running it with the option --allgen, including boards.txt.. Then you'll
> have to make a pull-request with all the modified files.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/esp8266/Arduino/discussions/7882#discussioncomment-373663>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIPMDFFU7PMSAZDIRUHT2OTS7JYXBANCNFSM4XWJLKHQ>
> .
>


-- 
*Foddis Gabriele*

+393403394922

**Reply by d-a-v** [reply]
> I need to clone https://github.com/esp8266/Arduino ???

First you must fork it on github, then yes it is required that you clone your fork locally to be able to create a pull-request for your board on this repository.

**Reply by Bighoneypot** [reply]
@d-a-v can I directly replace the board.txt.py file with my already created board.txt file? is this enough or is it necessary to edit other files? Thank you.
[Board.txt](https://github.com/esp8266/Arduino/files/5989338/Board.txt)

**Reply by d-a-v** [reply]
> @d-a-v can I directly replace the board.txt.py file with my already created board.txt file? 

Nope. The python script is designed to **rebuild** `boards.txt`.
Only few lines per board are needed. It is much more simple to modify `boards.txt.py` than to update `boards.txt` by hand.
The few lines on top of `boards.txt` aim at explaining that.

**Reply by Bighoneypot** [reply]
@d-a-v I need to modify the first line after #board list or I need to create and add new board on json??
Thank you!!
![modify first board](https://user-images.githubusercontent.com/35570068/108085491-aebc6980-7075-11eb-9458-f5d2927178c4.jpg)

**Reply by d-a-v** [reply]
A new one needs to be created.

**Comment by Bighoneypot** [comment]
Hi @d-a-v I have  problem with script python. I have this error.


Can you help me? Thank you!!
****************************************************
generated: ldscripts (in tools/sdk/ld/)
Traceback (most recent call last):
  File "tools/boards.txt.py", line 1992, in <module>
    all_boards()
  File "tools/boards.txt.py", line 1622, in all_boards
    print(id + optname + '=' + board['opts'][optname])
TypeError: 'set' object is not subscriptable
*********************************************************************

I have solved. I have send Pull request..
